crates/shim-sgx/src/handler/mod.rs

```diff
@@ -996,4 +1144,117 @@ impl<'a> Handler<'a> {
 
         Ok(())
     }
+
+    fn handle_page_fault(&mut self, addr: usize, error_code: PageFaultErrorCode) -> Result<(), ()> {
+        let tid = self.tcb.tid;
+        let addr = addr & !(Page::SIZE - 1);
```

Comment:
Is page size guaranteed to be a power of 2? If not, what happens here?

Reply:
yes, on x86_64 it's always 4096.